### PR TITLE
Added Kubectl tool installer task

### PR DESCRIPTION
**Pull request by Anumita**
Download and cache specific versions of kubectl.

### CODE REVIEW

File: Tasks/KubectlInstallerV0/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+"use strict"
+
+import tl = require("azure-pipelines-task-lib/task");
+import toolLib = require("azure-pipelines-tool-lib/tool");
+import kubectlutility = require("utility-common/kubectlutility");
+
+export async function getKuberctlVersion(): Promise<string> {
+    let kubectlVersion = tl.getInput("kubectlVersion");
+    if(kubectlVersion) {        
+        if (kubectlVersion === "latest" || kubectlVersion === "")
```
Comment (vithati) [inline Tasks/KubectlInstallerV0/src/utils.ts]: if (kubectlVersion === "latest") is enough.  If kubectlVersion  is empty  if(kubectlVersion) return false.

File: Tasks/KubectlInstallerV0/src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+"use strict"
+
+import tl = require("azure-pipelines-task-lib/task");
+import toolLib = require("azure-pipelines-tool-lib/tool");
+import kubectlutility = require("utility-common/kubectlutility");
+
+export async function getKuberctlVersion(): Promise<string> {
+    let kubectlVersion = tl.getInput("kubectlVersion");
```
Comment (vithati) [inline Tasks/KubectlInstallerV0/src/utils.ts]: do you want trim the spaces, kubetclversion.trim()?

File: Tasks/KubectlInstallerV0/task.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    "category": "Tool",
+    "visibility": [
+        "Build",
+        "Release"
+    ],
+    "preview": true,
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 0,
+        "Minor": 150,
```
Comment (vithati) [inline Tasks/KubectlInstallerV0/task.json]: 0.0.1

File: Tasks/KubectlInstallerV0/task.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        "Patch": 0
+    },
+    "demands": [],
+    "satisfies": [
+        "Kubectl"
+    ],
+    "minimumAgentVersion": "2.115.0",
```
Comment (vithati) [inline Tasks/KubectlInstallerV0/task.json]: is minimumAgentVersion required?

File: Tasks/KubectlInstallerV0/task.json (hunk 3 above)
Comment (Anumita) [reply Tasks/KubectlInstallerV0/task.json]: minor is now the sprint version
    and patch is the change made during the sprint

File: Tasks/KubectlInstallerV0/task.json (hunk 4 above)
Comment (Anumita) [reply Tasks/KubectlInstallerV0/task.json]: ill remove it

File: Tasks/KubectlInstallerV0/src/utils.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+// handle user input scenerios
+export function sanitizeVersionString(inputVersion: string) : string{
+    var version = toolLib.cleanVersion(inputVersion);
+    if(!version) {
+        throw new Error(tl.loc("NotAValidSemverVersion"));
+    }
+    
+    return "v"+version;
```
Comment (bansalaseem) [inline Tasks/KubectlInstallerV0/src/utils.ts]: should not we check whether it starts with v or not and based on that either add it or not.

File: Tasks/KubectlInstallerV0/src/utils.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+"use strict"
+
+import tl = require("azure-pipelines-task-lib/task");
+import toolLib = require("azure-pipelines-tool-lib/tool");
+import kubectlutility = require("utility-common/kubectlutility");
+
+export async function getKuberctlVersion(): Promise<string> {
+    let kubectlVersion = tl.getInput("kubectlVersion");
```
Comment (bansalaseem) [inline Tasks/KubectlInstallerV0/src/utils.ts]: Add debug logs

File: Tasks/KubectlInstallerV0/src/utils.ts (hunk 5 above)
Comment (Anumita) [reply Tasks/KubectlInstallerV0/src/utils.ts]: the above line "toollib.cleanVersion"  removes the v prefix

image

**Pull request merged**